src/main/java/com/google/alpollo/servlets/AutocompleteServlet.java

```diff
@@ -0,0 +1,77 @@
+package com.google.alpollo.servlets;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import com.google.alpollo.helpers.ConfigHelper;
+import com.google.api.client.http.GenericUrl;
+import com.google.api.client.http.HttpRequest;
+import com.google.api.client.http.HttpRequestFactory;
+import com.google.api.client.http.HttpResponse;
+import com.google.api.client.http.HttpTransport;
+import com.google.api.client.http.javanet.NetHttpTransport;
+import com.google.gson.Gson;
+import com.google.gson.JsonIOException;
+import com.google.gson.JsonSyntaxException;
+import com.jayway.jsonpath.JsonPath;
+import org.json.simple.JSONArray;
+import org.json.simple.JSONObject;
+import org.json.simple.parser.JSONParser;
+import org.json.simple.parser.ParseException;
+
+@WebServlet("/autocomplete")
+public class AutocompleteServlet extends HttpServlet {
+  private final Gson gson = new Gson();
+  private final static String SEARCH_STRING = "searchString";
+  private final static String TYPES = "types";
+
+  /**
+   * Making a POST request to this servlet with a search string and the types searched for
+   * as parameters will make a call to the Knowledge Graph Search API and it will return a list of
+   * possible search results.
+   * 
+   * The artist's name doesn't necessarily have to be complete
+   */
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String searchString;
+    String types;
```

Comment:
I think you didn't understand what I meant, client won't send to you types of entities, because it's not familiar with internal api and it shouldn't. I meant that type can be "artist" or "song"